integration/build_test.go

```diff
@@ -390,3 +389,29 @@ func getExternalIP(t *testing.T, c *NSKubernetesClient, ns string) string {
 	}
 	return ip
 }
+
+// Make sure GCB project is deduced from artifact's image name,
+// AFTER the default repo is applied.
+func TestBuildGCBWithDefaultRepo(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+	if !ShouldRunGCPOnlyTests() {
+		t.Skip("skipping test that is gcp only")
+	}
+
+	skaffold.Build("-d", "gcr.io/k8s-skaffold").InDir("testdata/gcb-default-repo").RunOrFail(t)
```

Comment:
At least a comment here would be nice that the reason we do not assert anything in this test is that it is assumed that the only way this build can succeed is if the right project id is determined for GCB. 

Now, having said that I would be most comfortable with explicitly test for the behavior you are introducing instead of indirectly testing it with integration tests. 

Still better than no tests :D

Reply:
Thanks, I'll add the comments. I tried to add some unit tests but it was a pain. I'll see what we can do to make that easier in the future.